wagtail/images/views/images.py

```diff
@@ -291,8 +292,11 @@ def preview(request, image_id, filter_spec):
     image = get_object_or_404(get_image_model(), id=image_id)
 
     try:
-        response = HttpResponse()
-        image = Filter(spec=filter_spec).run(image, response)
+        # Temporary image needs to be an instance that Willow can run optimizers on
+        temp_image = SpooledTemporaryFile(max_size=settings.FILE_UPLOAD_MAX_MEMORY_SIZE)
+        image = Filter(spec=filter_spec).run(image, temp_image)
+        temp_image.seek(0)
+        response = FileResponse(temp_image)
         response["Content-Type"] = "image/" + image.format_name
```

Comment:
Not a problem introduced by this PR, but shouldn't this last line be...
```suggestion
        response.headers["Content-Type"] = "image/" + image.format_name
```

Reply:
This is the newer syntax, yes. But using `response` like a `dict` still seems to be a supported pattern: https://docs.djangoproject.com/en/5.0/ref/request-response/#setting-header-fields. I don't think we need to deal with that in this PR.